packages/loot-core/src/client/accounts/accountsSlice.ts

```diff
@@ -0,0 +1,320 @@
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+
+import { send } from '../../platform/client/fetch';
+import { type AccountEntity, type TransactionEntity } from '../../types/models';
+import { addNotification, getAccounts, getPayees } from '../actions';
+import * as constants from '../constants';
+import { createAppAsyncThunk } from '../redux';
+import { type AppDispatch } from '../store';
+
+const sliceName = 'accounts';
+
+const initialState: AccountState = {
+  failedAccounts: {},
+  accountsSyncing: [],
+};
+
+type SetAccountsSyncingPayload = {
+  ids: Array<AccountEntity['id']>;
+};
+
+type MarkAccountFailedPayload = {
+  id: AccountEntity['id'];
+  errorType: string;
+  errorCode: string;
+};
+
+type MarkAccountSuccessPayload = {
+  id: AccountEntity['id'];
+};
+
+type AccountState = {
+  failedAccounts: {
+    [key: AccountEntity['id']]: { type: string; code: string };
+  };
+  accountsSyncing: Array<AccountEntity['id']>;
+};
+
+const accountsSlice = createSlice({
+  name: sliceName,
+  initialState,
+  reducers: {
+    setAccountsSyncing(
+      state,
+      action: PayloadAction<SetAccountsSyncingPayload>,
+    ) {
+      state.accountsSyncing = action.payload.ids;
+    },
+    markAccountFailed(state, action: PayloadAction<MarkAccountFailedPayload>) {
+      state.failedAccounts[action.payload.id] = {
```

Comment:
TIL, apparently in RTK it's [fine](https://redux-toolkit.js.org/usage/immer-reducers#redux-toolkit-and-immer) to mutate state now as long as you don't return anything!